### PR TITLE
`qml.QubitUnitary` jitting

### PR DESCRIPTION
**Pull request by albi3ro**
Fixes Issue #2247  [sc-15384]

This change makes it so that the unitary check is not performed if the matrix is abstract.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -148,6 +148,8 @@
 
 <h3>Bug fixes</h3>
 
+* The `QubitUnitary` now supports jitting. 
+
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:152]:
```suggestion
* The `qml.QubitUnitary` operation now supports jitting. 
  [(#2249)](https://github.com/PennyLaneAI/pennylane/pull/2249)
```

File: tests/ops/qubit/test_matrix_ops.py [hunk 2]
```diff
@@ -147,6 +147,16 @@ def test_qubit_unitary_jax(self, U, num_wires):
         with pytest.raises(ValueError, match="must be of shape"):
             qml.QubitUnitary(U, wires=range(num_wires + 1)).get_matrix()
 
+    @pytest.mark.parametrize("U, num_wires", [(H, 1), (np.kron(H, H), 2)])
+    def test_qubit_unitary_jax(self, U, num_wires):
```
Comment (antalszava) [inline tests/ops/qubit/test_matrix_ops.py:151]:
```suggestion
    def test_qubit_unitary_jax(self, U, num_wires):
        """Tests that QubitUnitary works with jitting."""
```

**Review by antalszava — APPROVED** [review]
Awesome fix! :1st_place_medal:

**Comment by ankit27kh** [comment]
@albi3ro I'm still getting an error with using JIT with qml.QubitUnitary.
The temp fix I'm using is to comment out the unitarity check in matrix_ops.py
I am on version 0.22.1

**Pull request merged**